### PR TITLE
Add --all-features flag to clippy

### LINKED ISSUE #9: pass --all-features to clippy
cargo-scout should be allowed to be passed --all-features and pass it to clippy

A live stream on how to fix #11, which almost requires the same steps as fixing this issue is available here https://youtube.com/watch?v=vrCuCnHYep0 :)

If someone would like to work on it, please let me know and I'll do anything I can to ease the process 🎉

### PR DESCRIPTION
**Pull request by undef1nd**
Closes #9

### CODE REVIEW

File: src/clippy.rs [hunk 1]
```diff
@@ -217,6 +227,24 @@ mod tests {
             no_default_features_command_parameters,
             no_default_features_linter.get_command_parameters()
         );
+
+        let all_features_linter = linter
```
Comment (o0Ignition0o) [inline src/clippy.rs:231]: There's even a test for the new flag, it's wonderful ! 🎉

**Review by o0Ignition0o — APPROVED** [review]
This is very solid work, thank you for doing it !
It looks great to me, and I'll merge it once it's not in draft anymore, and the CI passes :)
As previously discussed, passing --all-features as well as --no-default-features to clippy is fine, because we will let clippy figure out how to handle it, and we will behave as clippy does :)

File: src/clippy.rs (hunk 1 above)
Comment (undef1nd) [reply src/clippy.rs:231]: YAY! Putting it in Ready for Review. Thanks!

**Comment by o0Ignition0o** [comment]
And it's merged ! thank you for doing it !

**Pull request merged**